pkg/controller/postgresql.go

```diff
@@ -457,3 +466,40 @@ func (c *Controller) postgresqlDelete(obj interface{}) {
 
 	c.queueClusterEvent(pg, nil, spec.EventDelete)
 }
+
+/*
+  Ensures the service account required by StatefulSets to create pods exists in a namespace before a PG cluster is created there so that a user does not have to deploy the account manually.
+
+  The operator does not sync these accounts after creation.
+*/
+func (c *Controller) createPodServiceAccounts(event spec.ClusterEvent) error {
+
+	namespace := event.NewSpec.GetNamespace()
+	podServiceAccountName := c.opConfig.PodServiceAccountName
+
+	if _, ok := namespacesWithPodServiceAccount.Load(namespace); ok {
+		c.logger.Infof(fmt.Sprintf("The required pod service account %v already exists in the namespace %v", podServiceAccountName, namespace))
+		return nil
+	}
+
+	_, err := c.KubeClient.ServiceAccounts(namespace).Get(podServiceAccountName, metav1.GetOptions{})
+
+	if err != nil {
+
+		c.logger.Infof(fmt.Sprintf("creating pod service account in the namespace %v", namespace))
+		c.logger.Infof("the pod service account %q cannot be retrieved in the namespace %q. Trying to deploy the account.", podServiceAccountName, namespace)
+
+		// get a separate copy of service account
+		// to prevent a race condition when setting a namespace for many clusters
+		sa := *c.PodServiceAccount
+		_, err = c.KubeClient.ServiceAccounts(namespace).Create(&sa)
+		if err != nil {
+			return fmt.Errorf("cannot deploy the pod service account %q defined in the config map to the %q namespace: %v", podServiceAccountName, namespace, err)
+		}
+
+		namespacesWithPodServiceAccount.Store(namespace, podServiceAccountName)
+		c.logger.Infof("successfully deployed the pod service account %q to the %q namespace", podServiceAccountName, namespace)
+	}
```

Comment:
if the Get call returned us the service account we also need to store it in the namespacesWithPodServiceAccount map.